### DISCUSSION TITLE
is there a way to programmatically check that generated typescript client is up to date with the schema.prisma file?

### DISCUSSION THREAD

**Discussion by capaj** [prisma/prisma#19727] [Q&A] [answered]
as per title.

The reason I ask is because in development I often forget to generate it when I pull changes from remote and then I am dumbfounded as to why my app is not working. This would save me a lot of unnecessary trouble.

**Comment by ludralph** [comment]
Hi @capaj 👋 

Thank you for raising this question. At the moment, we do not have a way to programmatically check that the generated typescript client is up to date with the Schema file. However, we do have a feature request for this behaviour as documented here:

- https://github.com/prisma/prisma/issues/10977

There is a workaround that was shared by one of our community members within the same issue. Please check the link below:

- https://github.com/prisma/prisma/issues/10977#issuecomment-1116742619


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by capaj** [reply]
no this does not answer my question.
https://github.com/prisma/prisma/issues/10977#issuecomment-1116742619

only shows how to check that your DB is migrated to the last migration available.
I want to check that prisma TS client is generated up to date. These are two different things

**Comment by capaj** [comment, marked as answer]
For now I have worked around this problem by creating a custom prisma generator here: https://github.com/capaj/prisma-generator-checker

but it sure would be nice if this was part of prisma client by default or if at least prisma stored the schema hash in the generated client so that we can perform this check at runtime ourselves without the need for a custom generator.

**Reply by nikolasburk** [reply]
Wow, that's great — thanks for sharing the generator @capaj 🎉 

As for your other comment above: 

> I want to check that prisma TS client is generated up to date. These are two different things

Agreed! If you'd like to see this functionality natively, it would be really great if you could [open a new feature request ](https://github.com/prisma/prisma/issues/new?assignees=&labels=&projects=&template=feature_request.md&title=)so that our Product and Engineering can track this 🙂
